### PR TITLE
Add suport for local declarations.

### PR DESCRIPTION
**Pull request by rbartlensky**
The compiler now handles both global and local variables.

### CODE REVIEW

File: luacompiler/src/lib/irgen/mod.rs [hunk 1]
```diff
@@ -297,11 +338,14 @@ mod tests {
...
         let expected_instrs = vec![
-            HLInstr(Opcode::LDI, 1, 0, 0),
-            HLInstr(Opcode::LDS, 2, 0, 0),
-            HLInstr(Opcode::SetAttr, 0, 2, 1),
-            HLInstr(Opcode::LDS, 3, 1, 0),
-            HLInstr(Opcode::SetAttr, 0, 3, 1),
+            HLInstr(Opcode::LDI, 1, 0, 0),     // R(1) = INT(0) == 1
+            HLInstr(Opcode::LDS, 2, 0, 0),     // R(2) = STR(0) == "x"
+            HLInstr(Opcode::SetAttr, 0, 2, 1), // _ENV["x"] = R(1); x = 1
+            // XXX this is quite redundant because R(2) already has the value of "x"
```
[the remark this answers is not in the document]
Comment (rbartlensky) [reply luacompiler/src/lib/irgen/mod.rs]: I am loading into register 3 the index of "x" in the constant table, but I already have the value loaded into register 2. I left this comment to remind myself to go back to fix this. I will do this in another PR as it isn't as straight-forward as it seems.

[the remark this answers is not in the document]
Comment (ltratt) [reply luacompiler/src/lib/irgen/mod.rs]: Why isn't it straight-forward?

[the remark this answers is not in the document]
Comment (rbartlensky) [reply luacompiler/src/lib/irgen/mod.rs]: I can try to do it in this PR then. I just need to add a few more things to my register_map struct.

**Comment by rbartlensky** [comment]
I managed to remove the redundant load instructions. I think this is ready for another review.

**Comment by ltratt** [comment]
Please squash.

**Comment by ltratt** [comment]
Could you expand the commit message a bit? The way that you handle local variables is a bit fiddly, so some hints in the commit message would help a bit. Force push when you're done.

**Comment by rbartlensky** [comment]
Done. I think I manged to describe most of the things.

**Comment by ltratt** [comment]
Thanks!

**Pull request merged**